src/rabbit_mnesia.erl

```diff
@@ -785,6 +786,49 @@ check_nodes_consistency(Node, RemoteStatus = {RemoteAllNodes, _, _}) ->
                                         [node(), Node, Node])}}
     end.
 
+check_mnesia_consistency(Node, ProtocolVersion) ->
+    % If mnesia is running we will just check protocol version
+    % If it's not running, we don't want it to join cluster until all checks pass
+    % so we start it without `dir` env variable to prevent
+    % joining cluster and/or corrupting data
+    with_running_or_clean_mnesia(fun() ->
+        case negotiate_protocol([Node]) of
+            [Node] -> ok;
+            []     ->
+                LocalVersion = mnesia:system_info(protocol_version),
+                {error, {inconsistent_cluster,
+                         rabbit_misc:format("Mnesia protocol negotiation failed."
+                                            " Local version: ~p."
+                                            " Remote version ~p",
+                                            [LocalVersion, ProtocolVersion])}}
+        end
+    end).
+
+negotiate_protocol([Node]) ->
+    mnesia_monitor:negotiate_protocol([Node]).
+
+with_running_or_clean_mnesia(Fun) ->
+    MnesiaRunning = case mnesia:system_info(is_running) of
+        stopping ->
+            ensure_mnesia_not_running(),
+            no;
+        starting ->
+            ensure_mnesia_running(),
+            yes;
+        Other    -> Other
+    end,
+    case MnesiaRunning of
+        yes -> Fun();
+        no  ->
+            {ok, MnesiaDir} = application:get_env(mnesia, dir),
+            application:unset_env(mnesia, dir),
+            mnesia:start(),
```

Comment:
`{CWD}/Mnesia.{NODENAME}`
But it won't be created until schema with `disc_copies` is created. By default schema will have only `ram_copies`.